modules/auction_manager/src/lib.rs

```diff
@@ -690,3 +835,38 @@ impl<T: Trait> AuctionManager<T::AccountId> for Module<T> {
 		}
 	}
 }
+
+impl<T: Trait> EmergencyShutdown for Module<T> {
+	fn on_emergency_shutdown() {
+		Self::emergency_shutdown();
+	}
+}
+
+#[allow(deprecated)]
+impl<T: Trait> frame_support::unsigned::ValidateUnsigned for Module<T> {
+	type Call = Call<T>;
+
+	fn validate_unsigned(call: &Self::Call) -> TransactionValidity {
+		if let Call::cancel(auction_id) = call {
+			if !Self::is_shutdown() {
+				return InvalidTransaction::Stale.into();
+			} else if <CollateralAuctions<T>>::contains_key(auction_id) {
+				if !Self::collateral_auction_in_reverse_stage(*auction_id) {
+					return InvalidTransaction::Stale.into();
+				}
+			} else if !<SurplusAuctions<T>>::contains_key(auction_id) && !<DebitAuctions<T>>::contains_key(auction_id) {
+				return InvalidTransaction::Stale.into();
+			}
+
+			Ok(ValidTransaction {
+				priority: TransactionPriority::max_value(),
+				requires: vec![],
+				provides: vec![(<system::Module<T>>::block_number(), auction_id).encode()],
```

Comment:
no need to have block number in tag. we can't cancel an auction twice with different block numbers

add some unique byte to ensure the provides tag is unique.